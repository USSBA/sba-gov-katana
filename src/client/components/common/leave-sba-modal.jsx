import React from "react";
import Modal from 'react-modal'
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as ModalActions from '../../actions/show-modal.js'
import styles from '../../styles/modals/leave-sba-modal.scss'
import sbaLogo from '../../../../public/assets/images/logo.png';
import exitIcon from '../../../../public/assets/svg/exit-modal-close.svg';

class LeaveSbaModal extends React.Component {

  continueLink() {
    this.props.actions.closeLeaveSba();
    document.location = this.props.url
  }

  render() {
    return (
      <Modal isOpen={ true } className={ styles.content } overlayClassName={ styles.overlay }>
        <img className={ styles.logo } src={ sbaLogo } />
        <a onClick={ this.props.actions.closeLeaveSba }><img className={ styles.exitIcon } src={ exitIcon } /></a>
        <h2 className={ styles.title }> You're leaving the Small Business Administration website.</h2>
        <div className={ styles.divider }></div>
        <p className={ styles.text }>This link is provided for your reference only. The SBA doesn’t endorse non-government websites, companies, or applications. The SBA doesn’t attest to the accuracy
          of information provided by third-parties and other linked sites.</p>
        <div className={ styles.linkContainer }><span>Link to website: </span>
          <a className={ styles.link } tabIndex="0" onClick={ this.continueLink.bind(this)}>
            { this.props.url }
          </a>
        </div>
        <div className={ styles.btnContainer }>
          <button onClick={ this.props.actions.closeLeaveSba } className={ styles.btnCancel }>CANCEL
          </button>
          <button onClick={ this.continueLink.bind(this) } className={ styles.btnContinue }>CONTINUE</button>
        </div>
      </Modal>

      );
  }
}

function mapReduxStateToProps(reduxState) {
  return {
    url: reduxState.modalReducer.modalProps.targetUrl
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(ModalActions, dispatch)
  }
}
export default connect(
  mapReduxStateToProps,
  mapDispatchToProps
)(LeaveSbaModal);