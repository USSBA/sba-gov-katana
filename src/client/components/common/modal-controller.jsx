import React from 'react';
import { connect } from 'react-redux';
import LeaveSbaModal from "./leave-sba-modal.jsx";

const modalTypes = {
  "LEAVE_SBA": LeaveSbaModal
};

const ModalController = ({modalType, modalProps}) => {
  console.log(modalType);
  console.log(modalProps);
  if (!modalType) {
    return <div />
  } else {
    const ModalClass = modalTypes[modalType];
    console.log(ModalClass);
    return (<ModalClass {...modalProps} />);
  }
};

export default connect(
  (state) => {
    console.log(state.modalReducer)
    return state.modalReducer;
  }
)(ModalController);
