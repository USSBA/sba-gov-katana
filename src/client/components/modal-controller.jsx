import React from 'react'
import { connect } from 'react-redux'
import { SbaNewsModal } from 'molecules'
import { LeaveSbaModal, MobileSectionNavModal } from 'organisms'

const modalTypes = {
  LEAVE_SBA: LeaveSbaModal,
  SBA_NEWSLETTER: SbaNewsModal,
  MOBILE_SECTION_NAV: MobileSectionNavModal
}

const ModalController = ({ modalType, modalProps }) => {
  if (!modalType) {
    return <div />
  } else {
    const ModalClass = modalTypes[modalType]
    if (ModalClass) {
      return <ModalClass {...modalProps} />
    } else {
      return <div />
    }
  }
}

export default connect(state => {
  return state.modalReducer
})(ModalController)
