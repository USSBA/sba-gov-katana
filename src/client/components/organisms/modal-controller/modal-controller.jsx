import React from 'react'
import { connect } from 'react-redux'

import { SbaNewsModal } from 'molecules'
import { MobileSectionNavModal } from 'organisms'

const ModalController = ({ modalType, modalProps }) => {
  if (!modalType) {
    return <div />
  } else if (modalType === 'MOBILE_SECTION_NAV') {
    return <MobileSectionNavModal {...modalProps} />
  } else if (modalType === 'SBA_NEWSLETTER') {
    return <SbaNewsModal {...modalProps} />
  } else {
    return <div />
  }
}

export default connect(state => {
  return state.modalReducer
})(ModalController)

export { ModalController }
