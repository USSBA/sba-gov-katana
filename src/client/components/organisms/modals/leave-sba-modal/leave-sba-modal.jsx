import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import styles from './leave-sba-modal.scss'
import { Link } from 'atoms'
import { SbaModal } from 'molecules'
import * as ModalActions from '../../../../actions/show-modal.js'

class LeaveSbaModal extends React.Component {
  continueLink() {
    this.props.actions.closeLeaveSba()
    document.location = this.props.url
  }

  render() {
    const {
      actions: { closeLeaveSba },
      url
    } = this.props

    const title = "You're leaving the Small Business Administration website."
    const text =
      'This link is provided for your reference only. The SBA doesn’t endorse non-government websites, companies, or applications. The SBA doesn’t attest to the accuracy of information provided by third-parties and other linked sites.'

    return (
      <SbaModal
        showLogo={true}
        title={title}
        text={text}
        cancelButtonText="CANCEL"
        okButtonText="CONTINUE"
        onClose={closeLeaveSba}
        onClickOk={this.continueLink.bind(this)}
      >
        <div className={styles.linkContainer}>
          <span>Link to website:</span>
          <br />
          <Link to={url}>{url}</Link>
        </div>
      </SbaModal>
    )
  }
}

function mapReduxStateToProps(reduxState) {
  return { url: reduxState.modalReducer.modalProps.targetUrl }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(ModalActions, dispatch)
  }
}
export default connect(
  mapReduxStateToProps,
  mapDispatchToProps
)(LeaveSbaModal)

export { LeaveSbaModal }
