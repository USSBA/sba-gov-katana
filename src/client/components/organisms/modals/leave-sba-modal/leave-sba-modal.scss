.linkContainer {
    text-align: center;
    margin-bottom: 30px;
    font-size: 18px;

    @media screen and (max-width: 767px) {
        margin-bottom: 8vw;
        font-size: 4.3vw;
    }
}

.link {
    color: #0B97DD !important;
}

.link:hover {
    cursor: pointer;
}
