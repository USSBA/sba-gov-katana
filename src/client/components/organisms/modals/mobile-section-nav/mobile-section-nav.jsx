import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import style from './mobile-section-nav.scss'
import * as ModalActions from '../../../../actions/show-modal.js'
import { Link } from 'atoms'
import { navigateNow } from '../../../../services/navigation'
import { getLanguageOverride } from '../../../../services/utils.js'

class MobileNav extends React.Component {
  constructor(props) {
    super()
    this.state = {
      navMenu: 'open'
    }
    this._animationTimer = this._animationTimer.bind(this)
    this._handleBackBtn = this._handleBackBtn.bind(this)
    this._handleClick = this._handleClick.bind(this)
    this._navMenuClassname = this._navMenuClassname.bind(this)
  }

  _animationTimer() {
    setTimeout(() => {
      this.props.actions.closeMobileNav()
    }, 1000)
  }

  _handleBackBtn() {
    if (this.props.backUrl !== false) {
      this.props.actions.closeMobileNav()
      navigateNow(this.props.backUrl)
    } else {
      this.setState(
        {
          navMenu: 'close'
        },
        this._animationTimer
      )
    }
  }

  _handleClick(linkObject) {
    this.setState(
      {
        navMenu: 'close'
      },
      this._animationTimer
    )
    // navigateNow(linkObject.fullUrl)
  }

  _navMenuClassname() {
    if (this.state.navMenu === 'open') {
      return style.navMenuOpen
    } else if (this.state.navMenu === 'close') {
      return style.navMenuClose
    }
  }

  render() {
    const { menuData } = this.props
    const children = menuData.children || []
    const langCode = getLanguageOverride()
    let menuTitle
    if (menuData.spanishTranslation && langCode === 'es') {
      menuTitle = menuData.spanishTranslation.title
    } else {
      menuTitle = menuData.title
    }

    return (
      <div className={this._navMenuClassname()}>
        <div
          className={style.navHeader}
          onClick={() => {
            this._handleBackBtn()
          }}
        >
          <i className={style.navLeftArrow + ' fa fa-angle-left'} />
          <div className={style.navTitleContainer}>
            <h3 className={style.navTitle}>{menuTitle}</h3>
          </div>
        </div>
        <div className={style.navTopLine} />
        {children.map((linkObject, index) => {
          return (
            <NavLink key={index} link={linkObject} handleClick={this._handleClick} langCode={langCode} />
          )
        })}
      </div>
    )
  }
}

const NavLink = props => {
  const { link, langCode } = props
  let fullUrl
  let title
  if (props.link.spanishTranslation && langCode === 'es') {
    fullUrl = link.spanishTranslation.fullUrl
    title = link.spanishTranslation.title
  } else {
    fullUrl = link.fullUrl
    title = link.title
  }

  return (
    <Link
      className={style.navLink}
      id={props.iD}
      to={fullUrl}
      onMouseUp={() => {
        props.handleClick(link)
      }}
    >
      {/* <a */}
      {/*   id={props.iD} */}
      {/*   className={style.navLink} */}
      {/* > */}
      {title}
      {/* </a> */}
    </Link>
  )
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(ModalActions, dispatch)
  }
}

export default connect(
  null,
  mapDispatchToProps
)(MobileNav)

export { MobileNav }
