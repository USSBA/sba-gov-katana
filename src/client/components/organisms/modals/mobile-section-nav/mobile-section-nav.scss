.navMenu {
  position: fixed;
  z-index: 9999999999 !important;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: $sba-blue;
  display: block;

  @include grid-media($sba-grid-large) {
    display: none;
  }
}

.navMenuOpen {
  -webkit-animation: slideLeft 1s ease-out; /* Safari, Chrome and Opera > 12.1 */
  -moz-animation: slideLeft 1s ease-out; /* Firefox < 16 */
  -ms-animation: slideLeft 1s ease-out; /* Internet Explorer */
  -o-animation: slideLeft 1s ease-out; /* Opera < 12.1 */
  animation: slideLeft 1s ease-out;

  composes: navMenu;
  @keyframes slideLeft {
    from {
      margin-right: -4000px;
      margin-left: 2000px;
      opacity: 1;
    }
    to {
      margin-right: 0;
      margin-left: 0;
    }
  }
}

.navMenuClose {
  -webkit-animation: slideRight 1s ease-in forwards; /* Safari, Chrome and Opera > 12.1 */
  -moz-animation: slideRight 1s ease-in forwards; /* Firefox < 16 */
  -ms-animation: slideRight 1s ease-in forwards; /* Internet Explorer */
  -o-animation: slideRight 1s ease-in forwards; /* Opera < 12.1 */
  animation: slideRight 1s ease-in forwards;

  composes: navMenu;
  @keyframes slideRight {
    0% {
      margin-right: 0;
      margin-left: 0;
    }
    99% {
      margin-right: -222.222rem;
      margin-left: 111.111rem;

      opacity: 1;
    }
    100% {
      opacity: 0;
    }
  }
}

.navHeader {
  position: relative;
  display: block;
  height: 3.889rem;
}

.navTitleContainer {
  position: absolute;
  top: 60%;
  left: 3.111rem;
  display: block;
  transform: translateY(-50%);
}

// .navLargeTitle {
//   font-size: 1.889rem;
//   font-weight: 900;
//   margin: 0;
//   text-transform: uppercase;
//   color: $secondary-black-ish;
//   color: white;
// }

// .navSmallTitle {
//   font-size: 1.111rem;
//   font-weight: bold;
//   margin: 0;
//   text-transform: uppercase;
//   color: $secondary-black-ish;
//   color: white;
// }

.navTitle {
  margin: 0;
  color: white;
}

// .navRightArrow {
//   @include grid-media($sba-grid-small){
//     display: none !important;
//   }
//   @include grid-media($sba-grid-medium){
//     display: block !important;
//     font-size: 30px !important;
//     line-height: 30px !important;
//     color: white;
//     position: absolute;
//     top: 50%;
//     transform: translateY(-50%);
//     right: 20px;
//   }
// }

// .navLeftArrow {
//   font-size: 1.667rem !important;
//   line-height: 1.667rem !important;
//   position: absolute;
//   top: 50%;
//   left: 1.111rem;
//   transform: translateY(-50%);
//   color: white;
// }

.navLeftArrow {
  font-size: 1.667rem !important;
  line-height: 1.667rem !important;
  font-weight: bold;
  position: absolute;
  top: 60%;
  left: 1.722rem;
  transform: translateY(-50%);
  color: $sba-gray;
}

.navLinkContainer {
  position: relative;
  z-index: 9999 !important;
  padding: 0.556rem 1.944rem 0.556rem 1.667rem;
  cursor: pointer;
  border-bottom: 0.056rem solid rgba(255, 255, 255, 0.2);
}

.navLink {
  font-size: 1rem !important;
  font-weight: 300 !important;
  text-decoration: none !important;
  color: white !important;
}

.navTopLine {
  display: block;
  width: 100%;
  height: 0.056rem;
  margin: 0;
  margin-top: 1.111rem;
  background-color: rgba(255, 255, 255, 0.2);
}