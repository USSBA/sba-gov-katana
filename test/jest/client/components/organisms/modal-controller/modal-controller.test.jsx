import React from 'react'
import renderer from 'react-test-renderer'
import _ from 'lodash'
import { shallow, mount} from 'enzyme'

import { ModalController } from 'organisms/modal-controller/modal-controller.jsx'
import { MobileNav as MobileSectionNavModal } from 'organisms/modals/mobile-section-nav/mobile-section-nav.jsx'
import { LeaveSbaModal } from 'organisms/modals/leave-sba-modal/leave-sba-modal.jsx'


const modalInfo = {
  modalProps: {
    backUrl: false,
    icon: "build/img/img-ed52b1.png",
    menuData: {
      description: 'Partner with the SBA',
      fullUrl: 'partners/lenders',
      node: '2976',
      summary: 'Financial instituition',
      title: 'Lender',
      url: 'lenders',
      weight: -50,
      children: [
        {
          description: null,
          fulUrl: '/partners/lendrs/cdc-504-program',
          node: 2966,
          summary: 'If your organization is designated by the SBA',
          title: "CDC/504 program",
          url: "cdc-504-program",
          weight: -50,
          children: [
            {
              children: null,
              description: null,
              fullUrl: "partners/lenders/7a-loan-program/types-7a-loans",
              node: "2938",
              summary: "7(a) loan program",
              title: "Types of 7(a) loans",
              url: "types-of-7(a)",
              weight: -50
            },
            {
              children: null,
              description: null,
              fullUrl: "partners/lenders/7a-loan-program/community-advantage-loans",
              node: "2944",
              summary: "The SBA Community Advantage",
              title: "Community Loan",
              url: "community-advantage-loans",
              weight: -50
            }
          ]
        },
        {
          description: null,
          fulUrl: '/partners/lendrs/cdc-504-program',
          node: 2966,
          summary: 'If your organization is designated by the SBA',
          title: "CDC/504 program",
          url: "cdc-504-program",
          weight: -50,
          children: [
            {
              children: null,
              description: null,
              fulUrl: '/partners/lendrs/cdc-504-program/secondary-market',
              node: 2964,
              summary: 'There it is',
              title: "Secondary market",
              url: "secondary-market",
              weight: 0
            },
            {
              children: null,
              description: null,
              fulUrl: '/partners/lendrs/cdc-504-program/operate-cdc',
              node: 2956,
              summary: 'Streamlined the lending process',
              title: "Operate as a CDC",
              url: "operate-cdc",
              weight: 0
            }
          ]
        }
      ]
    }
  }
}

describe('ModalController', () => {
  test('should render mobile section navigation modalType', () => {
    const props = _.merge(modalInfo, { modalType: 'MOBILE_SECTION_NAV' })

    props.modalProps.store = {
      getState: () => {
        return false
      },
      subscribe: () => {
        return false
      },
      dispatch: () => {
        return false
      }
    }

  const component = mount(<ModalController {...props} />)
  expect(component.find('MobileNav')).toHaveLength(1)
  })

//   test('should render LeaveSba modalType', () => {
//     const props = {
//       modalType: 'LEAVE_SBA',
//       modalProps: { url: 'https://testing.com' }
//     }

//     props.modalProps.store = {
//       getState: () => {
//         return false
//       },
//       subscribe: () => {
//         return false
//       },
//       dispatch: () => {
//         return false
//       }
//     }

//     const component = mount(<ModalController {...props} />)
//     expect(component.find('LeaveSbaModal')).toHaveLength(1)
//   })
})