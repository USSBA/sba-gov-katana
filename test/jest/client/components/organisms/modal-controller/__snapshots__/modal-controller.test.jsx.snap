// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalController should render MobileSectionNav page 1`] = `
<div
  className="navMenuOpen"
>
  <div
    className="navHeader"
    onClick={[Function]}
  >
    <i
      className="navLeftArrow fa fa-angle-left"
    />
    <div
      className="navTitleContainer"
    >
      <h3
        className="navTitle"
      >
        Lender
      </h3>
    </div>
  </div>
  <div
    className="navTopLine"
  />
  <div
    className="navLinkContainer"
    onTouchTap={[Function]}
  >
    <a
      className="navLink"
      id={undefined}
      onTouchTap={[Function]}
    >
      CDC/504 program
    </a>
  </div>
  <div
    className="navLinkContainer"
    onTouchTap={[Function]}
  >
    <a
      className="navLink"
      id={undefined}
      onTouchTap={[Function]}
    >
      CDC/504 program
    </a>
  </div>
</div>
`;
